### PR TITLE
Ol6 select

### PR DESCRIPTION
**Pull request by drodenberg**
Resolves: attach_gh_issue_here

### PR Safety Checklist:

 - [ ] Added the task to the appropriate release doc under **Enhancements** or **Bug Fixes**
 - [ ] Bump `package.json` & `package-lock.json` version numbers to appropriate release
 - [ ] (optional) All external API changes have been documented
 - [ ] (optional) Build docs: `npm run docs`

### CODE REVIEW

File: src/Map/utils.js [hunk 1]
```diff
@@ -165,27 +187,23 @@ export function convertXYtoLatLong (map, x, y) {
...
+
+  if (opts?.map) {
+    const source = new VectorSource({ features: select.getFeatures() })
+    const vectorLayer = new VectorLayer({ style: layerStyle, source, map: opts.map })
+
+    vectorLayer.set('origin', layerName)
+    vectorLayer.set('_ol_kit_select', true)
+  }
+
+  console.warn('In order for the SelectInteraction to work properly the Map needs to be passed through the opts param.') // eslint-disable-line
```
Comment (drodenberg) [inline src/Map/utils.js:206]: move this to the else

Comment (drodenberg) [reply src/Map/utils.js:206]: back ticks around the `map`

File: src/Map/utils.js [hunk 2]
```diff
@@ -165,27 +187,23 @@ export function convertXYtoLatLong (map, x, y) {
...
-    })
-  })
 
-  return new olInteractionSelect({
-    hitTolerance: 3,
-    style: [DEFAULT_SELECT_STYLE],
-    ...props
-  })
+// TODO: This needs to take in a layername, select name, and styles and props so you can override anything
+export function createSelectInteraction (opts, selectName, layerName, selectStyle) {
```
Comment (drodenberg) [inline src/Map/utils.js:192]: opts and name for params

File: src/Popup/utils.js [hunk 3]
```diff
@@ -288,7 +269,9 @@ export const getPopupPositionFromFeatures = (event, features = [], opts = {}) =>
     return { arrow: 'none', pixel, fits: false }
   }
 
-  return getPosition(getFitsForFeatures(features))
+  const fitsForFeature = getFitsForFeatures(features)
```
Comment (drodenberg) [inline src/Popup/utils.js:272]: revert back

File: src/Map/utils.js [hunk 4]
```diff
@@ -165,27 +187,35 @@ export function convertXYtoLatLong (map, x, y) {
...
-    ...props
+    ...opts
   })
 }
+
+/**
+ * Create a new openlayers select interaction with default styling and add the vector layer to the map
+ * @function
+ * @category Map
+ * @since 0.2.0
```
Comment (stazrad) [inline src/Map/utils.js:202]: this needs to be `1.11.0`

File: src/Map/Map.js [hunk 5]
```diff
@@ -114,20 +115,27 @@ class Map extends React.Component {
...
+      // Layer panel also adds a select interaction
+      .filter(interaction => interaction._ol_kit_interaction_type !== '_ol_kit_layer_panel_hover')
+      // this checks if the select interaction created or passed in is the same instance on the map and never double adds
+      .find(interaction => interaction instanceof olInteractionSelect)
+
     if (selectInteraction) {
       // if select is passed as a prop always use that one first
       this.selectInteraction = selectInteraction
+    } else if (selectInteractionOnMap) {
+      this.selectInteraction = selectInteractionOnMap
```
Comment (drodenberg) [inline src/Map/Map.js:129]: add ugh warn to inform IA's that we are assigning internal select to this select interaction on the map

File: src/Map/Map.js [hunk 6]
```diff
@@ -114,20 +115,27 @@ class Map extends React.Component {
...
-    }
+      const { select, layer } = addSelectInteraction(map, 'ol_kit_map')
 
-    // check the map to see if select interaction has been added
-    const selectInteractionOnMap = map.getInteractions().getArray()
-      // Layer panel also adds a select interaction
-      .filter(interaction => interaction._ol_kit_interaction_type !== '_ol_kit_layer_panel_hover')
-      // this checks if the select interaction created or passed in is the same instance on the map and never double adds
-      .find(interaction => interaction === this.selectInteraction)
+      window.layer = layer
```
Comment (drodenberg) [inline src/Map/Map.js:134]: remove htis

File: src/Map/Map.js [hunk 7]
```diff
@@ -114,20 +115,27 @@ class Map extends React.Component {
...
+
+      this.selectInteraction = select
+      selectInteractionOnMap = true
+    }
 
     // do not double add the interaction to the map
     if (!selectInteractionOnMap) map.addInteraction(this.selectInteraction)
```
Comment (drodenberg) [inline src/Map/Map.js:141]: move this into the first if check

File: src/Map/Map.js (hunk 6 above)
Comment (drodenberg) [reply src/Map/Map.js:134]: add ugh warn to tell IA's what we made for them

File: src/Map/Map.js (hunk 5 above)
Comment (drodenberg) [reply src/Map/Map.js:129]: remove the else if and add a nice ugh warn message

File: src/Map/utils.js [hunk 8]
```diff
@@ -165,27 +187,35 @@ export function convertXYtoLatLong (map, x, y) {
...
+ * @returns {ol.interaction.Select} https://openlayers.org/en/latest/apidoc/module-ol_interaction_Select-Select.html
+ */
+export function addSelectInteraction (map, name, opts) {
+  const select = createSelectInteraction(opts)
+  const source = new VectorSource({ features: select.getFeatures() })
+  const layer = new VectorLayer({ source, style: [DEFAULT_SELECT_STYLE], map })
+
```
Comment (drodenberg) [inline src/Map/utils.js:209]: name = name || '_ol_kit_selection'

File: src/Popup/Popup.js [hunk 9]
```diff
@@ -116,6 +116,7 @@ class Popup extends Component {
     }, () => onMapClick(this.state))
 
     const layers = await Promise.all(promises)
+
```
Comment (drodenberg) [inline src/Popup/Popup.js:119]: remove this space

File: src/Popup/utils.js (hunk 3 above)
Comment (drodenberg) [reply src/Popup/utils.js:271]: nvm

File: src/Map/Map.js [hunk 10]
```diff
@@ -114,23 +115,28 @@ class Map extends React.Component {
...
       // if select is passed as a prop always use that one first
       this.selectInteraction = selectInteraction
+
+      // do not double add the interaction to the map
+      if (!selectInteractionOnMap) map.addInteraction(this.selectInteraction)
     } else {
       // otherwise create a new select interaction
-      this.selectInteraction = createSelectInteraction()
-    }
+      const { select } = addSelectInteraction(map, 'ol_kit_map')
```
Comment (stazrad) [inline src/Map/Map.js:133]: actual name prop does not match the ugh.warn 🤔

Comment (stazrad) [reply src/Map/Map.js]: solved this by passing nothing and just using the default 👌

File: src/Map/Map.js [hunk 11]
```diff
@@ -114,23 +115,28 @@ class Map extends React.Component {
...
-      // Layer panel also adds a select interaction
-      .filter(interaction => interaction._ol_kit_interaction_type !== '_ol_kit_layer_panel_hover')
-      // this checks if the select interaction created or passed in is the same instance on the map and never double adds
-      .find(interaction => interaction === this.selectInteraction)
+      ugh.warn('<Map> has created a default select interaction (use `getSelectInteraction(map)` to access it). To have ol-kit use your custom select interaction, pass `selectInteraction` as a prop to <Map>.')
 
-    // do not double add the interaction to the map
-    if (!selectInteractionOnMap) map.addInteraction(this.selectInteraction)
+      this.selectInteraction = select
+      selectInteractionOnMap = true
```
Comment (drodenberg) [inline src/Map/Map.js:138]: this can actually be removed now

**Comment by drodenberg** [comment]
LGTM

**Review by stazrad — APPROVED** [review]

**Pull request merged**